# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/FarhanNaseerAI/ML-FlyRank/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

# 1. My Lane (or Freestyle) and Why

I selected the **Refresh / Content Opportunity Scoring** lane.

The purpose of this project is to identify which content pages should be reviewed first for improvement. Large websites often contain thousands of pages, making it impractical to manually inspect every page. By analyzing observable search and engagement signals, machine learning can help prioritize pages that are most likely to benefit from review.

I selected this lane because it focuses on supporting real business decisions. Rather than replacing human judgment, the project will produce a ranked list of pages that helps content teams decide where to spend their limited time and resources.

In [8]:
import pandas as pd

# Load the dataset
df = pd.read_csv("/content/content_refresh_anonymized.csv")

# Display first 5 rows
df.head()

,content_id,client_id,search_volume,competition,competition_level,cpc,content_type,main_intent,word_count,char_count,...,char_count_tier,ctr,avg_position,engagement_rate,scroll_rate,ai_traffic_pct,impression_tier,position_tier,trend_direction,trend_pct
0,content_304f48230142,client_f369cb89fc,10.0,0.67,HIGH,2.05,keyword article,transactional,3221.0,20457.0,...,15000-25000,0.76,10.6,5.88,4.55,0.0,good,striking,down,-41.4
1,content_a1fb4e703a9e,client_4e07408562,90.0,0.01,LOW,0.05,keyword article,informational,2481.0,15562.0,...,15000-25000,0.05,20.3,0.00,10.00,0.0,good,page_3_5,down,-57.7
2,content_9aa793d4d895,client_7f2253d7e2,0.0,0.00,LOW,0.00,keyword article,informational,3515.0,23643.0,...,15000-25000,0.09,36.5,0.00,28.57,0.0,good,page_3_5,down,-60.9
3,content_331d6c4de07b,client_19581e27de,10.0,0.00,LOW,0.00,keyword article,commercial,NaN,NaN,...,NaN,0.49,6.2,1.28,3.45,0.0,good,page_1,stable,-13.8
4,content_d99b7a2d90ca,client_3fdba35f04,0.0,0.00,LOW,0.00,keyword article,informational,2803.0,17469.0,...,15000-25000,0.13,44.0,0.00,24.29,0.0,good,page_3_5,down,-34.7


In [9]:
print("Dataset Shape:", df.shape)

df.info()

Dataset Shape: (30000, 44)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30000 entries, 0 to 29999
Data columns (total 44 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   content_id              30000 non-null  object 
 1   client_id               30000 non-null  object 
 2   search_volume           27532 non-null  float64
 3   competition             27532 non-null  float64
 4   competition_level       27390 non-null  object 
 5   cpc                     27532 non-null  float64
 6   content_type            30000 non-null  object 
 7   main_intent             27626 non-null  object 
 8   word_count              22301 non-null  float64
 9   char_count              22301 non-null  float64
 10  provider_used           8562 non-null   object 
 11  model_used              24267 non-null  object 
 12  impressions_90d         30000 non-null  int64  
 13  clicks_90d              30000 non-null  int64  
 14  pageviews_9

# 2. The Question

### Research Question

Which content pages should be reviewed first to maximize opportunities for improving search performance?

### Decision

The decision is determining which content pages should receive attention first for updating, expanding, monitoring, or improving.

### Unit of Analysis

Each row in the dataset represents one content page.

### Output

The final output will be a ranked priority score that indicates how urgently each page should be reviewed.

### Action

Content editors can use the ranked list to review the highest-priority pages first instead of manually checking every page.

### Cost of a Wrong Recommendation

A false positive means a healthy page is reviewed unnecessarily, resulting in wasted time and effort.

A false negative means an important declining page is overlooked, potentially leading to continued traffic or engagement losses.

### Why Machine Learning?

The dataset contains multiple observable search and engagement signals such as impressions, clicks, sessions, CTR, average position, and engagement metrics. Machine learning can analyze the relationships among these signals and prioritize pages more efficiently than manual inspection alone.

In [10]:
# Basic dataset statistics

print("Rows:", df.shape[0])
print("Columns:", df.shape[1])

print("\nTrend Direction Distribution:")
print(df["trend_direction"].value_counts())

print("\nAverage 90-Day Impressions:", round(df["impressions_90d"].mean(), 2))
print("Average 90-Day Sessions:", round(df["sessions_90d"].mean(), 2))

Rows: 30000
Columns: 44

Trend Direction Distribution:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64

Average 90-Day Impressions: 5200.37
Average 90-Day Sessions: 37.07


# 3. Quick Look at the Data

The starter dataset contains **30,000 content pages** and **44 features** describing search performance, engagement, content characteristics, and trend information.

The trend distribution shows that **16,262 pages** are currently experiencing a downward trend, making declining content the largest category in the dataset. This suggests that content performance monitoring is an important problem worth investigating.

The dataset also reports an average of **5,200.37 impressions** and **37.07 sessions** over the last 90 days. These metrics indicate that the dataset contains meaningful search visibility and engagement information that can be used to prioritize pages for review.

These observations suggest that the dataset provides sufficient variation in content performance to support a machine learning approach for ranking content review opportunities.

# 4. Careful Words

This project is designed to support decision making rather than replace it.

The goal is to identify pages that appear to deserve review based on observable search and engagement signals. A high ranking does not guarantee that updating a page will improve its performance.

The results should be treated as recommendations that help content teams prioritize their work. Final decisions should always involve human review and additional context that may not be available in the dataset.

# 5. Self Check

- ✅ I selected one predefined project lane.
- ✅ I explained why I selected this lane.
- ✅ I clearly defined the research question.
- ✅ I identified the decision and action.
- ✅ I defined the unit of analysis.
- ✅ I explained the cost of incorrect recommendations.
- ✅ I supported my choice using real numbers from the dataset.
- ✅ I used careful language and avoided unsupported claims.